File: packages/web/src/types/game.ts
export interface Game {
  id: string
  title: string
  coverImageUrl: string | null
  minPlayers: number
  maxPlayers: number
  complexityWeight: number | null
  playTimeMinutes: number | null
  description: string | null
}

export interface TutorialStep {
  id: number
  phase: string
  title: string
  content: string
  imageUrl?: string | null
  estimatedSeconds: number
}

export interface TutorialData {
  gameId: string
  steps: TutorialStep[]
  totalSteps: number
  estimatedMinutes: number
}

export interface ReferenceItem {
  type: string
  title: string
  content: { title?: string; items: string[] }
  displayOrder: number
}

export interface ReferenceData {
  gameId: string
  references: ReferenceItem[]
}

export interface ChatChunk {
  text: string
}

export interface ChatDone {
  citations: Array<{ section?: string; chunkIndex?: number }>
  sessionId: string
}

export interface VisionResponse {
  analysis: string
  confidence: string
  relatedRules: string[]
}

export interface ChatMessage {
  id: string
  role: 'user' | 'assistant'
  content: string
  citations?: Array<{ section?: string }>
  isStreaming?: boolean
}

File: packages/api/app/schemas/game.py
from pydantic import BaseModel


class GameSummary(BaseModel):
    id: str
    title: str
    cover_image_url: str | None
    min_players: int
    max_players: int
    complexity_weight: float | None
    play_time_minutes: int | None
    description: str | None

    model_config = {"from_attributes": True}


class GameListResponse(BaseModel):
    games: list[GameSummary]


class TutorialStep(BaseModel):
    id: int
    phase: str
    title: str
    content: str
    image_url: str | None = None
    estimated_seconds: int = 60


class TutorialResponse(BaseModel):
    game_id: str
    steps: list[TutorialStep]
    total_steps: int
    estimated_minutes: int


class ReferenceItem(BaseModel):
    type: str
    title: str
    content: dict
    display_order: int


class ReferenceResponse(BaseModel):
    game_id: str
    references: list[ReferenceItem]


class ChatRequest(BaseModel):
    session_id: str | None = None
    message: str


class ChatChunk(BaseModel):
    text: str


class ChatDone(BaseModel):
    citations: list[dict] = []
    session_id: str


class VisionRequest(BaseModel):
    mode: str = "identify"
    session_id: str | None = None


class VisionResponse(BaseModel):
    analysis: str
    confidence: str
    related_rules: list[str] = []


class TTSRequest(BaseModel):
    text: str
    voice: str = "alloy"
